terraform/files/bin/deploy_cluster_api.sh

```diff
@@ -48,7 +48,8 @@ sudo mv calicoctl /usr/local/bin
 
 # Install cilium
 # TODO: Check signature
```

Comment:
Kind of unrelated: This TODO can be removed, right?

Reply:
Well downloading the binary and the cryptographic hash from the same place and checking that they match does not add any security against attackers that want to inject evil code into your deployment ... as both code and checksum come from the same source at the same time and are thus prone to the same potential attack.
It helps against corruption during transmission.
We would need to check cryptographic signatures (with a list of trusted keys that we retrieve beforehand and keep in the repo) or at least have a collection of known good cryptographic checksums stored in the repo and compare the downloaded binary ...

Reply:
Right. Made me think whether it'd be a good idea to create a GH Issue/Epic to decide on preferences regarding tooling/concepts providing such "Supply Chain Security" (like sigstore, TUF...), but that does not seem necessary, as there is a tender planned for lot 9b already.